[Li Yaner] iP

liy722 opened the pull request:
# DukePro

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://example.com))
DukePro free your mind of having to remember things you need to do. It's,

- textbased
- easy to learn
- ~~**FAST**~~ **SUPER FAST** to use

All you need to do is,

1. download it from [here](https://example.com).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
  
And it is *FREE*

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

___
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
       }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

liy722 committed:
Level-0: Rename

liy722 committed:
Level-1: Echo

liy722 committed:
Level-2: Add, List

liy722 committed:
Level-3: Mark as Done

liy722 committed:
Level-4: ToDos, Events, Deadlines

liy722 committed:
A-TextUiTesting: Automated Text UI Testing

liy722 committed:
Level-5: Handle Errors

liy722 committed:
Level-6: Delete

liy722 committed:
A-Enums: Use Enums

itsapaulblem commented on src/main/java/Todo.java:
```diff
@@ -0,0 +1,10 @@
+public class Todo extends Task{
```
Missing whitespace before the {

itsapaulblem commented on src/main/java/DukeException.java:
```diff
@@ -0,0 +1,5 @@
+class DukeException extends Exception {
```
class should be named AException as chatbot is named as A

itsapaulblem commented on src/main/java/A.java:
```diff
@@ -0,0 +1,94 @@
+                }
+
+                if (input.equals("list")) {
+                    if(count==0){
```
should have a whitespace before the {

GaaneshT replied:
Instead of count, I would suggest list.size(), it's more robust

itsapaulblem commented on src/main/java/A.java:
```diff
@@ -0,0 +1,94 @@
+                    System.out.println(" " + list.get(count));
+                    count++;
+                    System.out.println("Now you have " + count + " tasks in the list.");
+                }else if (input.startsWith("delete ")) {
```
Missing whitespace before the else if statement

GaaneshT replied:
For this, I think you can create a parser class and handle it there. You can also use a switch statement to make it more readable.

itsapaulblem commented on src/main/java/A.java:
```diff
@@ -0,0 +1,94 @@
+                        System.out.println("Now you have " + count + " tasks in the list.");
+                    }
+                }
+                else{
```
Missing whitespace after the else statement

itsapaulblem commented on src/main/java/A.java:
```diff
@@ -0,0 +1,94 @@
+                    } else {
+                        Task removedTask = list.remove(index);
+                        count--;
+                        System.out.println("Noted. I've removed this task:\n" + " "+removedTask);
```
missing whitespace before the variable removedTask

itsapaulblem commented on src/main/java/A.java:
```diff
@@ -0,0 +1,94 @@
+                else{
+                    throw new DukeException("please enter a valid task");
+                }
+            }catch (DukeException e) {
```
Missing whitespace before the catch statement

itsapaulblem reviewed (commented):
LGTM. Just some cosmetic changes.

GaaneshT commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,15 @@
+public class Deadline extends Task {
+
+    protected String by;
```
I think it's better to make by private, since it does not allow direct modification from child classes and does not violate encapsulation

GaaneshT reviewed (commented):
LGTM, a suggestion I can make would be to follow a more OOP-based approach, it would make it cleaner to read! Overall, good work so far! :)

liy722 committed:
Level-7: Save

liy722 committed:
Level-8: Dates and Times

liy722 committed:
A-MoreOOP: Use More OOP

liy722 committed:
A-Packages: Organize into Packages

liy722 committed:
Merge branch 'add-gradle-support' of https://github.com/liy722/ip

liy722 committed:
A-Gradle: Use Gradle

liy722 committed:
A-JUnit: Add JUnit Tests

liy722 committed:
A-JavaDoc: JavaDoc

liy722 committed:
A-CodingStandard: Follow the Coding Standard

liy722 committed:
Level-9: find

liy722 committed:
A-CheckStyle: Use CheckStyle optional

liy722 committed:
Level-10: GUI

liy722 committed:
A-Assertions: Implement A-Assertions to improve code reliability

liy722 committed:
Merge pull request #2 from liy722/A-Assertions

A-Assertions: Implement A-Assertions to improve code reliability

liy722 committed:
Refactor code to improve quality and readability

liy722 committed:
Merge pull request #3 from liy722/A-CodeQuality

A-CodeQuality

liy722 committed:
add fixedDurationTask

liy722 committed:
Merge pull request #4 from liy722/Add-extension

add fixedDurationTask

liy722 committed:
A-UserGuide: User Guide

liy722 committed:
Integrate Checkstyle to enforce coding standards

The current codebase lacks a consistent coding standard, making it
difficult to maintain and read. Variations in code style can lead to
misunderstandings and increase the complexity of the code review
process.

Checkstyle is a development tool that helps programmers adhere to a
coding standard by checking Java source code against a set of
configuration rules.

Let's integrate Checkstyle into the build process by:
* Adding a Checkstyle configuration file based on our preferred coding
  standards.
* Updating the build scripts to include Checkstyle checks.
* Fixing existing code to adhere to the defined coding standards.

This integration will ensure a uniform code style across the project,
improving readability and maintainability.

A-CodeQualityIntegrate Checkstyle to enforce coding standards

The current codebase lacks a consistent coding standard, making it
difficult to maintain and read. Variations in code style can lead to
misunderstandings and increase the complexity of the code review
process.

Checkstyle is a development tool that helps programmers adhere to a
coding standard by checking Java source code against a set of
configuration rules.

Let's integrate Checkstyle into the build process by:
* Adding a Checkstyle configuration file based on our preferred coding
  standards.
* Updating the build scripts to include Checkstyle checks.
* Fixing existing code to adhere to the defined coding standards.

This integration will ensure a uniform code style across the project,
improving readability and maintainability.

A-CodeQuality

liy722 committed:
no message

liy722 committed:
Merge commit '2003cb55434b1fac3a3ee0a87182cead154e89ae'

liy722 committed:
Introduce assertions to validate input parameters in utility functions

Several utility functions in the codebase assume that input parameters
are valid, but do not explicitly verify these assumptions. This can lead
to unexpected behavior and errors when invalid inputs are passed.

Assertions can be used to enforce these assumptions during development,
helping to catch bugs early and ensuring that functions are used
correctly.

Let's add assertions to critical utility functions to:
* Check for non-null inputs.
* Verify that numeric parameters fall within expected ranges.
* Ensure that collection inputs are not empty when required.

These assertions will serve as a form of documentation and improve code
reliability by making the code's assumptions explicit.

liy722 committed:
correct build.gradle

liy722 committed:
v0.4

liy722 committed:
v0.5

liy722 committed:
style: Unify code indentation and formatting

liy722 committed:
style: Unify code indentation and formatting

liy722 committed:
style: Unify code indentation and formatting

liy722 committed:
refactor: Remove unused imports

liy722 committed:
refactor: Make field `a` final
